[Outlook](legacy tokens) Revised timeline for turning off Exchange tokens (#5017)
OfficeDev/office-js-docs-pr: docs/outlook/turn-exchange-tokens-on-off.md

ms.service: microsoft-365
ms.subservice: add-ins
ms.topic: how-to
ms.date: 02/03/2025
---

# Turn legacy Exchange Online tokens on or off

> Legacy Exchange Online tokens are deprecated and will begin being turned off across Microsoft 365 tenants on February 17th, 2025. If you are a developer migrating your Outlook add-in from legacy tokens to Entra ID tokens and nested app authentication, you'll need to test updates to your add-in. You can use the Exchange Online PowerShell cmdlets to turn legacy tokens on or off. Turn off legacy tokens in a test tenant to confirm that your updated Outlook add-in is working correctly.

Reviewer: ```suggestion
Legacy Exchange Online tokens are deprecated and will be turned off across Microsoft 365 tenants starting February 17th, 2025. If you're a developer migrating your Outlook add-in from legacy tokens to Entra ID tokens and nested app authentication, you'll need to test updates to your add-in. Use the Exchange Online PowerShell cmdlets to turn off legacy tokens in a test tenant to confirm that your updated Outlook add-in is working correctly.
```

> [!NOTE]
> The previous command is the only way to view legacy token status. Other commands, such as `Get-AuthenticationPolicy | Format-Table -Auto Name` don't return the legacy token status.

This command only shows the legacy token status as set by the administrator. If the administrator has never changed the settings, the command returns `(Not Set)`. If the token status is `(Not Set)` when the February deployment by Microsoft to turn off legacy tokens is implemented, the token status will still be `(Not Set)` even though legacy tokens are off. The following table shows the behavior of legacy Exchange Online tokens based on the token status when the February change is applied.

> | Legacy token status  | Legacy token status after February change  | Legacy token behavior after February change |

Reviewer 2: added comments about this offline...
